test/TradePageBloc_test.dart

```diff
@@ -0,0 +1,37 @@
+import 'package:ax_dapp/dialogs/buy/usecases/GetSwapInfoUseCase.dart';
+import 'package:ax_dapp/pages/trade/bloc/TradePageBloc.dart';
+import 'package:ax_dapp/service/BlockchainModels/SwapInfo.dart';
+import 'package:ax_dapp/service/Controller/Swap/SwapController.dart';
+import 'package:ax_dapp/service/Controller/WalletController.dart';
+import 'package:mockito/annotations.dart';
+import 'package:test/test.dart';
+
+import 'TradePageBloc_test.mocks.dart';
+
+@GenerateMocks([GetSwapInfoUseCase, WalletController, SwapController])
+void main() {
+  test("Should Calculate the Metrics Correctly on the Trade Page ", () async {
+    final mockUseCase = MockGetSwapInfoUseCase();
+    final mockWalletController = MockWalletController();
+    final mockSwapController = MockSwapController();
+    final tradePage = TradePageBloc(
+        repo: mockUseCase,
+        walletController: mockWalletController,
+        swapController: mockSwapController);
+    final swapInfo = SwapInfo(toReserve: 1, fromReserve: 7500, toPrice: 0.0001333333333333333333333333333333333);
+    final double inputAmount = 1;
+
+    final result = tradePage.calculateTransactionInfo(
+        swapInfo, inputAmount);
+    print(result.receiveAmount!.toStringAsFixed(17));
+    print(result.priceImpact!.toStringAsFixed(2));    
+    print(result.minimumReceived!.toStringAsFixed(8));
+    print(result.totalFee!.toStringAsFixed(6));
+    print(result.price!.toStringAsFixed(6));
+    assert(result.receiveAmount!.toStringAsFixed(17) == "0.00013291566441101");
+    assert(result.priceImpact!.toStringAsFixed(2) == "0.03");
+    assert(result.minimumReceived!.toStringAsFixed(8) == "0.00013159");
+    assert(result.totalFee!.toStringAsFixed(6) == "0.003000");
+    assert(result.price!.toStringAsFixed(6) == "0.000133");
+  });
+}
```

Comment:
This test was already done so is not very useful since its basically the same test as in buyDialogBloc. We are going to abstract this logic to a delegate class so we can do these calculates once and for all. 

I was asking you to put a test for the refresh function. If you see my latest test for [BuyDialogBloc](https://github.com/SportsToken/ax_dapp/pull/337/files#diff-bc80af3b746f56fd9e5ffb8456f4fb98bca491143e3f67fd76c1201995a43b33) I have a good example there on how you can test these bloc functions for my onMaxBuyTap test

Reply:
Okay I compare the test, I see that the swap calculation is different in a few ways so this is fine. 
We can make a ticket to add a test for the refreshEvent